core/tests/test_api.py

```diff
@@ -0,0 +1,234 @@
+import pytest
+from django.test.client import Client
+from django.urls import reverse
+
+from conftest import CommunityWithMembersFixture, CreateCommunitiesFixture, create_posts
+from core.models import Community, Post
+
+pytestmark = pytest.mark.django_db
+
+
+PAGE_SIZE = 10
+PREFIX = "http://testserver"
+
+
+def get_abs_url(url: str, page: int) -> str:
+    return f"{PREFIX}{url}?page={page}"
+
+
+visible_privacies = [Community.PUBLIC, Community.RESTRICTED]
+
+
+def three_pages_pagination(url: str, client: Client) -> None:
+    page_1_response = client.get(url)
+    page_2_response = client.get(page_1_response.data["next"])
+    page_3_response = client.get(page_2_response.data["next"])
+
+    endpoint_abs_url = f"{PREFIX}{url}"
+    page_1_abs_url = get_abs_url(url, page=1)
+    page_2_abs_url = get_abs_url(url, page=2)
+    page_3_abs_url = get_abs_url(url, page=3)
+
+    assert page_1_response.status_code == 200
+    assert page_2_response.status_code == 200
+    assert page_3_response.status_code == 200
+    assert len(page_1_response.data["results"]) == PAGE_SIZE
+    assert len(page_2_response.data["results"]) == PAGE_SIZE
+    assert len(page_3_response.data["results"]) == 1
+    assert page_1_response.data["previous"] is None
+    assert page_1_response.data["next"] == page_2_abs_url
+    assert page_2_response.data["previous"] in (endpoint_abs_url, page_1_abs_url)
+    assert page_2_response.data["next"] == page_3_abs_url
+    assert page_3_response.data["previous"] == page_2_abs_url
+    assert page_3_response.data["next"] is None
+    assert page_1_response.data["count"] == 2 * PAGE_SIZE + 1
+    assert page_2_response.data["count"] == 2 * PAGE_SIZE + 1
+    assert page_3_response.data["count"] == 2 * PAGE_SIZE + 1
+
+
+@pytest.mark.parametrize("privacy", visible_privacies)
+def test_api_communities_list_three_pages(
+    client: Client, create_communities: CreateCommunitiesFixture, privacy: str
+) -> None:
+    create_communities(count=PAGE_SIZE * 2 + 1, posts_per_community=0, privacy=privacy)
+    url = reverse("api-communities-list")
+    three_pages_pagination(url, client)
```

Comment:
I would change that name to something more meaningfull
```suggestion
    validate_three_pages_pagination(url, client)
```
or similar. I love to see "action" in the name.